### PR TITLE
OCPBUGS-42535: when no type is specified, don't make illegal condition

### PR DESCRIPTION
**Pull request by deads2k**

### CODE REVIEW

File: pkg/controllers/routercerts/controller.go [hunk 1]
```diff
@@ -100,6 +100,11 @@ func (c *routerCertsDomainValidationController) sync(ctx context.Context, syncCt
 	// set the condition anywhere in sync() to update the controller's degraded condition
 	var condition operatorv1.OperatorCondition
 	defer func() {
+		if len(condition.Type) == 0 {
+			// no change is desired.  This happens when the SyncPartialSecret fails
```
Comment (p0lyn0mial) [inline pkg/controllers/routercerts/controller.go:104]: yeah, in line `123`

File: pkg/controllers/routercerts/controller.go [hunk 2]
```diff
@@ -100,6 +100,11 @@ func (c *routerCertsDomainValidationController) sync(ctx context.Context, syncCt
 	// set the condition anywhere in sync() to update the controller's degraded condition
 	var condition operatorv1.OperatorCondition
 	defer func() {
+		if len(condition.Type) == 0 {
```
Comment (p0lyn0mial) [inline pkg/controllers/routercerts/controller.go:103]: how about adding a helper function for checking if a condition is empty ?

Comment (deads2k) [reply pkg/controllers/routercerts/controller.go:103]: to save a length check?  No.

**Pull request merged**